library/data/registry-entry-attributes/src/main/java/org/quiltmc/qsl/registry/attribute/impl/DumpBuiltinAttributesCommand.java

```
@@ -0,0 +1,192 @@
+		}
+
+		@Override
+		public Registry<?> parse(StringReader reader) throws CommandSyntaxException {
+			var id = Identifier.fromCommandInput(reader);
+			return Registry.REGISTRIES.getOrEmpty(id).orElseThrow(() -> UNKNOWN_REGISTRY_EXCEPTION.create(id));
+		}
+
+		@Override
+		public Collection<String> getExamples() {
+			return EXAMPLES;
+		}
+
+		@Override
+		public <S> CompletableFuture<Suggestions> listSuggestions(CommandContext<S> context, SuggestionsBuilder builder) {
+			return CommandSource.suggestIdentifiers(Registry.REGISTRIES.getIds(), builder);
+		}
+	}
+
+	public static void register(CommandDispatcher<ServerCommandSource> dispatcher) {
+		dispatcher.register(literal("dumpbuiltinregattrs")
+				.then(argument("registry", RegistryArgumentType.registry())
+						.requires(src -> src.hasPermissionLevel(4))
+						.executes(DumpBuiltinAttributesCommand::execute))
+		);
+	}
+
+	private static final Logger LOGGER = LogManager.getLogger("DumpBuiltinAttributesCommand");
+	private static final SimpleCommandExceptionType IO_EXCEPTION =
+			new SimpleCommandExceptionType(new LiteralText("IO exception occurred, check logs"));
+	private static final SimpleCommandExceptionType ILLEGAL_STATE =
+			new SimpleCommandExceptionType(new LiteralText("Encountered illegal state, check logs"));
+	private static final SimpleCommandExceptionType ENCODE_FAILURE =
+			new SimpleCommandExceptionType(new LiteralText("Failed to encode value, check logs"));
+	private static final SimpleCommandExceptionType UNCAUGHT_EXCEPTION =
+			new SimpleCommandExceptionType(new LiteralText("Uncaught exception occurred, check logs"));
+
+	private static int execute(CommandContext<ServerCommandSource> ctx) throws CommandSyntaxException {
+		var registry = RegistryArgumentType.getRegistry(ctx, "registry");
+		try {
+			execute0(ctx, registry);
+		} catch (RuntimeException e) {
+			LOGGER.error("Uncaught exception occurred", e);
+			throw UNCAUGHT_EXCEPTION.create();
+		}
```

Won't `CommandManager` catch all exceptions for you anyway?

---

AFAIK, it just swallows the exception and doesn't print the stacktrace - catching it manually allows me to do that.